add pull_request_template.md

jbonnell-amd opened the pull request:
A pull request template was recently added to compute (https://github.com/ROCm/rocprofiler-compute/pull/729), so this will add a similar pull request template to the systems repo.

I have copied the template below, which will now be used when creating a new PR in this repo.

# rocprofiler-systems Pull Request

## Related Issue
<!-- Please link to the issue(s) that this PR addresses.  -->
- [ ] Closes #<issue number or link>

## What type of PR is this? (check all that apply)

- [ ] Bug Fix
- [ ] Cherry Pick
- [ ] Continuous Integration
- [ ] Documentation Update
- [ ] Feature
- [ ] Optimization
- [ ] Refactor
- [ ] Other (please specify)

## Technical Details
<!-- Please explain the changes. -->

## Have you added or updated tests to validate functionality?

- [ ] Yes
- [ ] No - does not apply to this PR

## Added / Updated documentation?

- [ ] Yes
- [ ] No - does not apply to this PR

## Have you updated CHANGELOG?
<!-- Needed for Release updates for a ROCm release. -->
- [ ] Yes
- [ ] No - does not apply to this PR

jbonnell-amd committed:
add pull_request_template.md

dgaliffiAMD commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,27 @@
+<!-- Needed for Release updates for a ROCm release. -->
+- [ ] Yes
+- [ ] No - does not apply to this PR
+
```
```suggestion
## Added / Updated documentation?
- [ ] Yes
- [ ] No - does not apply to this PR

## Have you updated CHANGELOG?
<!-- Needed for Release updates for a ROCm release. -->
- [ ] Yes
- [ ] No - does not apply to this PR

```

jbonnell-amd committed:
Update .github/PULL_REQUEST_TEMPLATE/pull_request_template.md

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

jbonnell-amd committed:
fix markdown linting errors

sputhala-amd reviewed (dismissed):
LGTM

dgaliffiAMD commented:
Are there any concerns from @ROCm/rocm-documentation, @prbasyal-amd?

dgaliffiAMD commented on .github/PULL_REQUEST_TEMPLATE/pull_request_template.md:
```diff
@@ -0,0 +1,33 @@
+
+## What type of PR is this? (check all that apply)
+
+- [ ] Refactor
+- [ ] Feature
+- [ ] Bug Fix
+- [ ] Optimization
+- [ ] Documentation Update
+- [ ] Continuous Integration
+- [ ] Other (please specify)
```
```suggestion
- [ ] Bug Fix
- [ ] Cherry Pick
- [ ] Continuous Integration
- [ ] Documentation Update
- [ ] Feature
- [ ] Optimization
- [ ] Refactor
- [ ] Other (please specify)
```
I've added "Cherry-Pick" to the list of types and sorted alphabetically.

jbonnell-amd committed:
Update .github/PULL_REQUEST_TEMPLATE/pull_request_template.md

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

jbonnell-amd commented:
Will need approval from someone in @ROCm/rocm-documentation before being able to merge.

jbonnell-amd committed:
moved pull_request_template.md to base /.github/ directory

jbonnell-amd commented:
I noticed that the PR template I added to compute was not showing up when creating a new PR. Having looked into this, it seems like we need to add query parameters to create a pull request if using the PULL_REQUEST_TEMPLATE subdirectory, which is mostly used for cases where you have multiple templates you want to use.

- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request

Since we're just using the one, I moved the `pull_request_template.md` file to the base `.github` directory since it should detect this without needing query parameters.

jbonnell-amd commented:
Needs approval from @ROCm/rocm-documentation before being able to merge
